FWCore/Framework/test/stubs/TestStreamAnalyzers.cc

```diff
@@ -552,6 +554,263 @@ namespace edmtest {
       }
     };
 
+    class TestInputProcessBlockCache {
+    public:
+      long long int value_ = 0;
+    };
+
+    class TestInputProcessBlockCache1 {
+    public:
+      long long int value_ = 0;
+    };
+
+    class InputProcessBlockIntAnalyzer
+        : public edm::stream::EDAnalyzer<
+              edm::InputProcessBlockCache<int, TestInputProcessBlockCache, TestInputProcessBlockCache1>> {
+    public:
+      explicit InputProcessBlockIntAnalyzer(edm::ParameterSet const& pset) {
+        {
+          expectedByRun_ = pset.getParameter<std::vector<int>>("expectedByRun");
+          sleepTime_ = pset.getParameter<unsigned int>("sleepTime");
+          auto tag = pset.getParameter<edm::InputTag>("consumesBeginProcessBlock");
+          if (not tag.label().empty()) {
+            getTokenBegin_ = consumes<IntProduct, edm::InProcess>(tag);
+          }
+        }
+        {
+          auto tag = pset.getParameter<edm::InputTag>("consumesEndProcessBlock");
+          if (not tag.label().empty()) {
+            getTokenEnd_ = consumes<IntProduct, edm::InProcess>(tag);
+          }
+        }
+        registerProcessBlockCacheFiller<TestInputProcessBlockCache1>(
+            getTokenBegin_,
+            [this](edm::ProcessBlock const& processBlock,
+                   std::shared_ptr<TestInputProcessBlockCache1> const& previousCache) {
+              auto returnValue = std::make_shared<TestInputProcessBlockCache1>();
+              returnValue->value_ += processBlock.get(getTokenBegin_).value;
+              returnValue->value_ += processBlock.get(getTokenEnd_).value;
+              return returnValue;
+            });
+      }
+
+      static void accessInputProcessBlock(edm::ProcessBlock const&) {
+        edm::LogAbsolute("InputProcessBlockIntAnalyzer") << "InputProcessBlockIntAnalyzer::accessInputProcessBlock";
+      }
+
+      void analyze(edm::Event const& event, edm::EventSetup const&) override {
+        auto cacheTuple = processBlockCaches(event);
+        if (!expectedByRun_.empty()) {
+          if (expectedByRun_[event.run()] !=
+              std::get<edm::CacheHandle<TestInputProcessBlockCache1>>(cacheTuple)->value_) {
+            throw cms::Exception("UnexpectedValue")
+                << "InputProcessBlockIntAnalyzer::analyze cached value was "
+                << std::get<edm::CacheHandle<TestInputProcessBlockCache1>>(cacheTuple)->value_
+                << " but it was supposed to be " << expectedByRun_[event.run()];
+          }
+        }
+        // Force events to be processed concurrently
+        if (sleepTime_ > 0) {
+          usleep(sleepTime_);
```

Comment:
Seems reasonable to move towards a solution in std:: instead of a C/POSIX header if it does the same thing. I can see 39 instances of unistd.h in the Core code I have checked out, so it also seems reasonable to do this in the future  in a different PR, so I will leave this as is for now.

Reply:
Done. I changed them to use STATIC_REQUIRE. That should be better.